File: src/main/java/ssg/search/collection/es/bshop/ESBshopItem.java
package ssg.search.collection.es.bshop;

import com.google.common.collect.HashMultimap;
import com.google.common.collect.ImmutableSet;
import com.google.common.collect.Lists;
import com.google.common.collect.Multimap;
import org.apache.commons.lang3.StringUtils;
import ssg.search.collection.es.ESQuery;
import ssg.search.collection.es.disp.ESItem;
import ssg.search.constant.Shpp;
import ssg.search.parameter.Parameter;

import java.util.List;
import java.util.Map;

public class ESBshopItem extends ESItem {
	@Override
	public String getCollectionAliasName(Parameter parameter) {
		return "ESBshopItem";
	}

	@Override
	public List<String> getQuery(Parameter parameter) {
		Multimap<String, String> queryMap = HashMultimap.create();

		for (ESQuery query : ImmutableSet.of(
				ESQuery.SRCH_PREFIX_BSHOP
				, ESQuery.DISP_CTG_LST
                , ESQuery.DISP_CTG_SUB_LST
				, ESQuery.SALESTR_LST
				, ESQuery.DEVICE_CD
				, ESQuery.MBR_CO_TYPE
				, ESQuery.BSHOP_ID
		)) {
			if (StringUtils.isNotEmpty(query.getValue(parameter))) {
				queryMap.put(query.getField(parameter), query.getValue(parameter));
			}
		}

		String strShpp = StringUtils.defaultIfEmpty(parameter.getShpp(), "");
		String strPickuSalestr = StringUtils.defaultIfEmpty(parameter.getPickuSalestr(), "");
		if (strShpp.indexOf("picku") > -1) {
			if (!strPickuSalestr.equals("")) {
				queryMap.put("SHPP", Shpp.PICKU.getPrefix() + strPickuSalestr);
			} else {
				queryMap.put("SHPP", Shpp.PICKU.getPrefix());
			}
		}

		List<String> queryList = Lists.newArrayList();
		for (Map.Entry<String, String> e : queryMap.entries()) {
			if (StringUtils.isNotEmpty(e.getValue())) {
				queryList.add(e.getKey() + ":(" + e.getValue() + ")");
			}
		}
		return queryList;
	}

}

File: src/main/kotlin/ssg/front/search/api/collection/es/bshop/EsBshopItem.kt
package ssg.front.search.api.collection.es.bshop

import com.google.common.collect.HashMultimap
import ssg.front.search.api.collection.es.EsQuery
import ssg.front.search.api.collection.es.common.EsItem
import ssg.front.search.api.dto.Parameter

class EsBshopItem : EsItem() {
    override fun getQuery(parameter: Parameter): List<String> {
        val queryMap = HashMultimap.create<String, String>()

        for (query in listOf(
                EsQuery.SRCH_PREFIX_BSHOP
                , EsQuery.DISP_CTG_LST
                , EsQuery.DISP_CTG_SUB_LST
                , EsQuery.SALESTR_LST
                , EsQuery.DEVICE_CD
                , EsQuery.MBR_CO_TYPE
                , EsQuery.BSHOP_ID
        )) {
            if (query.getValue(parameter).isNotEmpty()) {
                queryMap.put(query.getField(parameter), query.getValue(parameter))
            }
        }

        val strShpp = parameter.shpp ?: ""
        val strPickuSalestr = parameter.pickuSalestr ?: ""
        if (strShpp.indexOf("picku") > -1) {
            if (strPickuSalestr != "") {
                queryMap.put("SHPP", "PICKU$strPickuSalestr")
            } else {
                queryMap.put("SHPP", "PICKU")
            }
        }

        val queryList = mutableListOf<String>()
        for ((key, value) in queryMap.entries()) {
            if (!value.isNullOrEmpty()) {
                queryList.add("$key:($value)")
            }
        }

        return queryList
    }
}